roles/manage-gcp-infra/tasks/cleanup.yml

```diff
@@ -68,6 +68,16 @@
   register: deployment
   failed_when: deployment.rc|int > 1
   changed_when: false
+
+- name: Check registry bucket for files
+  shell: gsutil ls -l gs://{{ google_registry_bucket_name }}
+  register: gs_file_count
+
+- name: Empty registry bucket contents
+  shell: gsutil rm gs://{{ google_registry_bucket_name }}/**
+  when:
+  - deployment.stdout
```

Comment:
Since `deployment` is being checked here, I presume it's for knowing that a deployment exists? If so, probably shave have the same check for the "Check registry bucket for files" above? Preferably just group these in a `block` checking for a valid deployment. Also, is any `stdout` categorized as a deployment exists? ... or is there a better check to be done here? I.e.: I see line 69 relies on a return code - maybe that's better here too?

Reply:
@oybed I've updated this to block the cluster cleanup components together. From what I'm seeing ( and using), any stdout from both `deployment` and `gs_file_count` show that there is a valid deployment and existing files in the storage bucket. 

On another note, it looks like there's an existing module to deal with gcp storage. It just looks like it relies on tokens versus being able to use the json file that we're using now. It may be worth while to see if we can get this to work with the json file or open up an issue with the ansible folks to see if we can improve these modules. I'd much rather be able to lean on modules versus having to rely on the right packages/versions to be installed on a system for this to work.